client/src/components/Announcements.js

```diff
@@ -10,13 +14,61 @@ class AnnouncementList extends Component {
     constructor(props) {
       super(props);
       this.state = {
-        announcements : this.props.announcements,
+        announcements : [],
           id: this.props.id
       }
+      
+    }
+
+    /**
+     * Makes a GET request to either dev or prod and fetches the announcement list
+     * as soon as the component loads and updates the state
+     * TODO: Add Validation for output
+     * TODO: Once finished, remove debugging statements
+     */
+    componentDidMount() {
+        if (this.ReactIsInDevelomentMode()) {
```

Comment:
The only difference between these two conditional statements is `urlToFetch`, I would simplify it to:

```
var urlToFetch = API_ANNOUNCEMENTS_PROD
if (this.ReactIsInDevelomentMode()) {
    urlToFetch = API_ANNOUNCEMENTS_DEV;
}
fetch(urlToFetch)
            .then(response => response.json())
            .then(data =>  this.setState({announcements:this.parseAnnouncements(data)}))
```